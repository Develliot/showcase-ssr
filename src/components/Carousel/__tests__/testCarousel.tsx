import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Carousel, CarouselLeftButton, CarouselRightButton } from '../';

it('renders correctly', () => {
    const wrapper: ShallowWrapper = shallow(
        <Carousel>
            {['a', 'b', 'c'].map((item, index) => {
                return <div key={index}>{item}</div>;
            })}
        </Carousel>
    );
    expect(wrapper).toMatchSnapshot();
});

it('renders navigates correctly', () => {
    const wrapper: ShallowWrapper = shallow(
        <Carousel>
            {['a', 'b', 'c'].map((item, index) => {
                return <div key={index}>{item}</div>;
            })}
        </Carousel>
    );

    expect(wrapper.find(CarouselRightButton)).toHaveLength(1);
    wrapper.find(CarouselRightButton).simulate('click');
    expect(wrapper.find(CarouselLeftButton)).toHaveLength(1);
    wrapper.find(CarouselRightButton).simulate('click');
    expect(wrapper.find(CarouselRightButton)).toHaveLength(0);
    wrapper.find(CarouselLeftButton).simulate('click');
    expect(wrapper).toMatchSnapshot();
});
