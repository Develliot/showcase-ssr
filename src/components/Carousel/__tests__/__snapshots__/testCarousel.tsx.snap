// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Fragment>
  <styles__CarouselWrapper>
    <styles__CarouselScrollArea
      onScroll={[Function]}
    >
      <styles__CarouselCard
        key="0"
      >
        <styles__CarouselCardInner
          isSelected={true}
        >
          <div
            key=".$0"
          >
            a
          </div>
        </styles__CarouselCardInner>
      </styles__CarouselCard>
      <styles__CarouselCard
        key="1"
      >
        <styles__CarouselCardInner
          isSelected={false}
        >
          <div
            key=".$1"
          >
            b
          </div>
        </styles__CarouselCardInner>
      </styles__CarouselCard>
      <styles__CarouselCard
        key="2"
      >
        <styles__CarouselCardInner
          isSelected={false}
        >
          <div
            key=".$2"
          >
            c
          </div>
        </styles__CarouselCardInner>
      </styles__CarouselCard>
    </styles__CarouselScrollArea>
    <styles__CarouselFadeOutLeft
      visible={false}
    />
    <styles__CarouselFadeOutRight
      visible={true}
    />
    <CarouselRightButton
      onClick={[Function]}
    />
  </styles__CarouselWrapper>
  <Layout__VerticalSpacer
    size="small"
  />
  <Dots
    count={3}
    currentPosition={0}
    scrollToPosition={[Function]}
  />
</Fragment>
`;

exports[`renders navigates correctly 1`] = `
<Fragment>
  <styles__CarouselWrapper>
    <styles__CarouselScrollArea
      onScroll={[Function]}
    >
      <styles__CarouselCard
        key="0"
      >
        <styles__CarouselCardInner
          isSelected={false}
        >
          <div
            key=".$0"
          >
            a
          </div>
        </styles__CarouselCardInner>
      </styles__CarouselCard>
      <styles__CarouselCard
        key="1"
      >
        <styles__CarouselCardInner
          isSelected={true}
        >
          <div
            key=".$1"
          >
            b
          </div>
        </styles__CarouselCardInner>
      </styles__CarouselCard>
      <styles__CarouselCard
        key="2"
      >
        <styles__CarouselCardInner
          isSelected={false}
        >
          <div
            key=".$2"
          >
            c
          </div>
        </styles__CarouselCardInner>
      </styles__CarouselCard>
    </styles__CarouselScrollArea>
    <styles__CarouselFadeOutLeft
      visible={true}
    />
    <styles__CarouselFadeOutRight
      visible={true}
    />
    <CarouselLeftButton
      onClick={[Function]}
    />
    <CarouselRightButton
      onClick={[Function]}
    />
  </styles__CarouselWrapper>
  <Layout__VerticalSpacer
    size="small"
  />
  <Dots
    count={3}
    currentPosition={1}
    scrollToPosition={[Function]}
  />
</Fragment>
`;
